src/java.base/share/classes/java/util/zip/ZipFile.java

```diff
@@ -1342,14 +1361,15 @@ private static boolean isZip64ExtBlockSizeValid(int blockSize) {
             /*
              * As the fields must appear in order, the block size indicates which
              * fields to expect:
+             *  0 - May be written out by Ant and Apache Commons Compress Library
```

Comment:
I don't like that `isZip64ExtBlockSizeValid()` still accepts `0` as *valid* input. I think we should fully handle the zero case in `checkZip64ExtraFieldValues()` (also see my comments there).

Reply:
I don't think this is a question of "taste" because `isZip64ExtBlockSizeValid()` suggests that the method will check for *valid* sizes and to my understanding `0` is not a valid input. This method might also be called from other places in the future which do not handle the zero case appropriately.

In any case, I'm ready to accept this as a case of "Disagree and Commit"  :) but in that case please update at least the comment below to something like "*..Note we do not need to check blockSize is >= 8 as we know its length is at least 8 by now*" because "*..from the call to isZip64ExtBlockSizeValid()*" isn't true any more.

Reply:
I think I agree with Volker that it would be better if isZip64ExtBlockSizeValid continued to return false for block size 0.

Reply:
OK, I have made the suggest change that you both prefer.

Thank you for your input

Reply:
I'm also happy to see `isZip64ExtBlockSizeValid` rejecting 0. This logic could be useful when implementing support for valid Zip64 fields for small entries in ZipInputStream, like #12524 attempted to do.  (The PR was closed by the bots in the end). 

I guess this method could be moved to ZipUtils if JDK-8303866 is ever implemented.